src/catchup/ApplyCheckpointWork.cpp

```diff
@@ -243,21 +222,18 @@ ApplyLedgerChainWork::onRun()
             openCurrentInputFiles();
         }
 
-        if (!applyHistoryOfSingleLedger())
-        {
-            mCurrSeq += mApp.getHistoryManager().getCheckpointFrequency();
-            mFilesOpen = false;
-        }
-
-        mApp.getCatchupManager().logAndUpdateCatchupStatus(true);
-
+        auto result = applyHistoryOfSingleLedger();
```

Comment:
I do not understand the change:

* either we change `applyHistoryOfSingleLedger` to always throw on error. The only time it returns `false` is when it has a bad input file (EOF) and it looks like it would go on an infinite loop with the updated logic
* or we handle the `false` case like before, and force the file to be reopen, but it looks to me that this is useless (what do we expect to be different with the retry?)

Reply:
Not sure what you are referring to: returning `false` does mean `EOF`, in which case work decides whether it succeeded or failed based on the expected LCL, see line 247 (otherwise, how would it pass any catchup test that applies multiple checkpoints?)

Also, why would the files need to be re-open in `onRun`? Any failure will invoke `onReset`, which already takes care of it.

There's a distinction here between throwing and returning false: the former means that something got unexpectedly corrupted, while the latter means that we've read everything in this file.